recurring_contract/model/recurring_contract.py

```diff
@@ -343,8 +342,7 @@ def contract_active(self, cr, uid, ids, context=None):
         return True
 
     def contract_terminated(self, cr, uid, ids, context=None):
-        today = datetime.today().strftime(DF)
-        self.write(cr, uid, ids, {'state': 'terminated', 'end_date': today})
+        self.write(cr, uid, ids, {'state': 'terminated'})

```

Comment:
You can make this change in module sponsorship_compassion as well (and you are not obliged to call super() from the inherited method, to avoid calling this write).